### PR TITLE
[WIP] Aggregation Extension

### PR DESCRIPTION
**Pull request by philvarner**
This is a early draft of a spec for an aggregation extension to solicit feedback. This is similar to the Elasticsearch aggregation api and to the similar /aggregate endpoint (https://eod-catalog-svc-prod.astraea.earth/aggregate) that Astraea Earth OnDemand implements. 

**Related Issue(s):** #

**Proposed Changes:**

1. 
2. 

**PR Checklist:**

- [ ] This PR has **no** breaking changes.
- [ ] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-api-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [ ] API only: I have run `npm run generate-all` to update the [generated OpenAPI files](https://github.com/radiantearth/stac-api-spec/blob/dev/README.md#openapi-definitions).

### CODE REVIEW

**Comment by m-mohr** [comment]
We spoke a bit about this on the sprint and think /aggregation should just return an extended version of a STAC Collection.

File: extensions/aggregation/README.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
+# Aggregation Extension
+
+The purpose of the Aggregation Extension is to provide an endpoint similar to the Search endpoint (`/search`), but which will provide aggregated information on matching Items rather than the Items themselves. This is useful when a dataset is very large, and it is infeasible to example all results.  This is particularly useful in data exploration, whereby a user of the data can change queries to see the "shape" of the results for a given query.
```
Comment (gsf-sellis) [inline extensions/aggregation/README.md:3]: "example all results" should maybe be "examine all results?"

File: extensions/aggregation/README.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+| intersects   | GeoJSON Geometry | Searches items by performing intersection between their geometry and provided GeoJSON geometry.  All GeoJSON geometry types must be supported. |
+| ids          | \[string]        | Array of Item ids to return. All other filter parameters that further restrict the number of search results (except `next` and `limit`) are ignored |
+| collections  | \[string]        | Array of Collection IDs to include in the search for items. Only Items in one of the provided Collections will be searched |
+| aggregations | \[string]        | A list of aggregations to compute and return |
+
```
Comment (gsf-sellis) [inline extensions/aggregation/README.md:31]: This table appears to be a duplicate of the table here? https://github.com/radiantearth/stac-api-spec/tree/master/item-search#query-parameters-and-fields
For maintenance purposes, I wonder if maybe we want to just reference that table here?

**Comment by philvarner** [comment]
Note to self: add `/aggregatables` endpoint

**Comment by philvarner** [comment]
Closing this PR in favor of putting this in separate repo. This PR https://github.com/philvarner/aggregation/pull/1/files how has these changes.

**Pull request closed without merging**